### PR TITLE
Add Counterstrike mission Sarin Gas 1: Crackdown

### PR DESCRIPTION
**Pull request by Smittytron**
I took what [forum user SethXXX](https://forum.openra.net/viewtopic.php?f=83&t=20812) started and tried to get it up to par. I'm skeptical of what I did with the TruckEscape routes so let me know if there's a way to combine those functions.

Counterstrike missions raise some questions on what to do regarding originality by using or not using Aftermath units. This mission and at least one Soviet Counterstrike mission uses tesla tanks but with the mobile radar jammer skin. I just went with the tesla tank here.

I also altered the starting units, starting cash, and allowed construction of mechanics for balance reasons.
We can hash out how close to the original these missions should be, but my two cents are that as the expansion missions have an unknown relationship with the timeline of the vanilla campaign, we can be more liberal with what units we allow the player to build.

### CODE REVIEW

**Comment by Nineeva** [comment]
Where can I download this?

**Comment by Nineeva** [comment]
Thanks,

**Comment by Muh-Muh** [comment]
I have not really looked at the code but played through it.   Just found a couple of minor things.

The following should not be buildable by allies (because they can not build a tech centre):  
Gap generator  
Chronosphere  
Cruisers  
(Also perhaps Ants & Zombies. I never played the original mission but I feel like being able to build these is probably not intended)  
The (allied) nuke should probably also only become available if you build a tech center after you have captured the enemy conjard.  

Another thing I didn't quite get is why there are so many submarines owned by Turkey which is neutral to the player. Does their existence have a point?

Otherwise no issues found while playing the mission

As for coding efficency, why not simply use one move truck function and one send truck function?
```
MoveTruck = function(truck, direction)
	Media.DisplayMessage("Convoy truck attempting to escape!")
	Media.PlaySoundNotification(greece, "AlertBleep")
	Utils.Do(direction, function(waypoint)
		truck.Move(waypoint.Location)
	end)
end
```
You can then call this function with (for example):
```
MoveTruck(truck1, MoveTruckNorth)
```
Then the sendTruck function can be called by:

```
SendTruck = function(truck, direction)
	if truck.IsDead then
		return
	else
		MoveTruck(truck, direction)
	end
end
```

and finally the triggers look like
```
Trigger.AfterDelay(DateTime.Minutes(5), SendTruck(truck1, MoveTruckNorth))
```
or something alone these lines

**Comment by Smittytron** [comment]
> Another thing I didn't quite get is why there are so many submarines owned by Turkey which is neutral to the player. Does their existence have a point?

Not sure what they are in for but the subs are in the original map. Possibly just in for flavor.

Updated. I addressed the rest of the above comments and added a commit to remove ant/zombie production options in campaign-rules.

**Comment by MustaphaTR** [comment]
> Another thing I didn't quite get is why there are so many submarines owned by Turkey which is neutral to the player. Does their existence have a point?

Looking at the original map ini, it seems like Turkey should be enemy to the player, not neutral.

**Comment by MustaphaTR** [comment]
I'm not sure if i like the changes to the starting units, original starts with 2 Spies, 1 Minelayer and 5 Rifles. That's too different from what we have here. Also they come from out of the map, not start them preplaced.
Also You start with Mediums even tho you can't build them.
Tho, i kinda feel like starting unit could be adjusted according to difficulty.

Another thing, it seems like Advanced Power Plant wasn't available in this mission in the original. Tho considering it is T2 in OpenRA, while it was available after building a normal one. Maybe changing that isn't necessary.

File: mods/ra/maps/sarin-gas-1-crackdown/rules.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	Buildable:
+		Prerequisites: ~vehicles.soviet
+
+V2RL:
+	Buildable:
+		Prerequisites: ~vehicles.soviet
+
+TTNK:
+	Buildable:
+		Prerequisites: ~vehicles.soviet
```
Comment (MustaphaTR) [inline mods/ra/maps/sarin-gas-1-crackdown/rules.yaml]: IMO this should keep Tesla Coil as a prereq.

File: mods/ra/maps/sarin-gas-1-crackdown/map.yaml [hunk 2]
```diff
@@ -0,0 +1,1066 @@
...
+		AllowBots: False
+		Playable: True
+		Required: True
+		LockFaction: True
+		Faction: allies
+		LockColor: True
+		Color: ABB7E4
+		LockSpawn: True
+		LockTeam: True
+		Enemies: USSR, BadGuy, Ukraine
```
Comment (MustaphaTR) [inline mods/ra/maps/sarin-gas-1-crackdown/map.yaml]: You forgot to make Turkey enemy of Greece.

**Comment by Muh-Muh** [comment]
One last request, other than that I think the mission plays just fine.
Could you change the stance of the turkey subs? Right now they are on return fire and don't really do anything. I assume that they should be  a little bit more aggressive.

**Comment by Smittytron** [comment]
Updated with original starting units and re-seasoned for balance. I adjusted the second wave of reinforcements with different units per difficulty level; with hard having the original three rockets and one light tank.

The original starts the player with 3700, which is a problem for scaling. I set it at the standard 5000.

Other comments addressed as well.

**Comment by reaperrr** [comment]
We should try to get this into the playtest.

**Comment by reaperrr** [comment]
I haven't finished testing this yet, but one thing that immediately caught my attention:

Even on 'Easy', the start is ridiculously harsh even compared to the original's 'Hard' (which I watched on youtube).
2 out of the 4 minigunners escorting the retreating truck were retreating with it in the original, and the minigunner from the north and grendiers from the south would arrive *much* later, giving you enough time to kill the 2 remaining escort minigunners first.

tl;dr: In all 3 attempts on Easy so far, my minigunners and spy got killed quickly with at least 4-5 enemy soldiers left, with the north minigunner then later also killing one and wounding another e3 from the 2nd wave as well, because I couldn't build a barracks and kill the enemies off that fast.

In my opinion, at least on 'Easy' the start should not be harder than in the original's 'Hard', meaning a) 2 of the 4 escort minigunners should run away or at least stay away far enough to not attack, and b) both the minigunner from north and grenadiers from south should arrive at least some 15-20 seconds later (and preferrably not at the same time, either), instead of all enemies from all directions coming in at once. That way, even slower players like me should be able to at least save the spy and kill more of the attackers.

**Comment by Smittytron** [comment]
Updated. Two rifles now retreat with the truck and the three soldiers on the outside will wait ten seconds before attacking.

**Comment by reaperrr** [comment]
With the Turkey subs adressed, I consider Muh-Muh's comment a +1.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/0eef15f76e4ee37ee9cda977a4b32093a7e2c2cf...035d79f9398d46ba56b9917758949f2403ccfe13)
